components/desktop/mate/mate-session-manager/mate-session-manager.p5m

```diff
@@ -26,6 +26,7 @@ license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
 
 depend type=require fmri=gnome/gnome-keyring
 depend type=require fmri=desktop/mate/marco
+depend fmri=pkg:/driver/graphics/nvidia fmri=pkg:/driver/graphics/nvidia-340 fmri=pkg:/driver/graphics/nvidia-470 fmri=pkg:/x11/library/mesa fmri=pkg:/driver/graphics/nvidia-390 type=require-any
```

Comment:
I believe it is a bad idea for MATE to depend directly and explicitly on a display driver.

Reply:
This is not just the display driver but also the whole opengl stack which is nvidia driver dependant. Check out ogl-select for how this has been dealt with since OpenSolaris

Reply:
What will happen if MATE is run without NVIDIA hardware?  What added value is dependency on `nvidia` then?

Reply:
nvidia is in those cases installed aswell due to dependencies. However int that case and if anvidia is not installed mesa is picked which delivers the opengl stack for OSS graphics drivers. Checkout ogl-select what we switch between at the moment.

Reply:
BTW, the `x11/server/xorg/driver/xorg-video` already depends on `nvidia` packages.  Yes, you added `mesa` to the list here, but still, shouldn't instead be `x11/server/xorg/driver/xorg-video`  modified accordingly (i.e. to add `mesa` to the list there together with `nvidia` packages)?

Reply:
xorg-video driver are drivers. we depend on mesa because of libGL and libEGL. They have nothing to do with xorg-video drivers. Nvidia delivers it's own patched version of libGL and EGL and we thus have to depend on all possible versions of lib[E]GL. I do not see why we should introduce artificial dependencies on Xorg packages we do not need to depend on here.

Reply:
> xorg-video driver are drivers. we depend on mesa because of libGL and libEGL. They have nothing to do with xorg-video drivers. Nvidia delivers it's own patched version of libGL and EGL and we thus have to depend on all possible versions of lib[E]GL. I do not see why we should introduce artificial dependencies on Xorg packages we do not need to depend on here.

This explanation is what I missed.  Thank you!

Maybe something similar (stripped down, reworded) re libGL and libEGL should be added as a comment directly to the manifest so it is there for future reference.

For the actual code, please remove `pkg:/` from all fmris you added to the manifest.  It will cause that these packages will not appear in the `REQUIRED_PACKAGES` list.  I believe this is what you need here.  Also it would be great if the list of `require-any` packages is sorted somehow (maybe with one package per line?).

Thank you.

Reply:
I can sort it but the pkg thing is not what is wrong. These packages cannot be installed simultaneously thus we have to remove them from the REQUIRED_PACKAGES so we don't kill people's dev setups.

Reply:
The `pkg:/` thing is exactly what had to be removed (as you did) because when there are fmris with `pkg:/` in manifests then such packages are added automatically to `REQUIRED_PACKAGES`.  If fmris in manifests are without `pkg:/` then such packages are NOT added automatically to `REQUIRED_PACKAGES`.